### PR TITLE
feat: adapt ui to routes

### PR DESCRIPTION
**Pull request by noahjoeris**
This PR
- adds chain and token ordering based on routes
- resets dest and token on source chain change
- fixes issue of token deselect on swap
- adds some cleanups

### CODE REVIEW

**Review by NunoAlexandre — COMMENTED** [review]
We shouldn't reset the token and dest chain when changing the source as long as the route is still supported. 

What the rationale?

**Comment by noahjoeris** [comment]
fixed

File: app/src/utils/filters.ts [hunk 1]
```diff
@@ -62,7 +63,11 @@ export const getAllowedTokens = (
     }
   })
 
-  return tokens
+  return orderByAllowedTag(tokens) as (Token & { allowed: boolean })[]
```
Comment (Victor-Salomon) [inline app/src/utils/filters.ts:66]: should we get this (Token & { allowed: boolean }) as a proper reusable type ?

File: app/src/hooks/useTransferForm.tsx [hunk 2]
```diff
@@ -104,38 +100,38 @@ const useTransferForm = () => {
...
+
+      if (!newValue || newValue.uid === sourceChain?.uid) return
+
+      const isSameDestination = destinationChain?.uid === newValue.uid
+
+      if (
+        destinationChain &&
+        tokenAmount &&
+        !isSameDestination &&
+        isRouteAllowed(environment, destinationChain, newValue, tokenAmount)
```
Comment (Victor-Salomon) [inline app/src/hooks/useTransferForm.tsx:113]: I will soon PR a better version of this helper. I will ensure that I'm fixing any potential issues/conflicts.

File: app/src/utils/filters.ts (hunk 1 above)
Comment (noahjoeris) [reply app/src/utils/filters.ts:66]: We could. But the question is, will we use it again in the future

File: app/src/hooks/useTransferForm.tsx (hunk 2 above)
Comment (noahjoeris) [reply app/src/hooks/useTransferForm.tsx:113]: Amazing. Yeah I don't like my way that much.

File: app/src/utils/filters.ts (hunk 1 above)
Comment (Victor-Salomon) [reply app/src/utils/filters.ts:66]: You are probably right. Probably not so let's not get annoyed by this anyway

File: app/src/hooks/useTransferForm.tsx (hunk 2 above)
Comment (Victor-Salomon) [reply app/src/hooks/useTransferForm.tsx:113]: I was speaking about the `isRouteAllowed` helper only 😅 (I already drafted the PR)

This one largely does what it's supposed to do so we will soon move on 🦾

**Pull request merged**